src/renderers/webgl/WebGLPrograms.js

```diff
@@ -1,6 +1,6 @@
 THREE.WebGLPrograms = function ( renderer, capabilities ) {
 
-	var programs = [];
+	var programs = renderer.info.programs;
```

Comment:
Or better

```javascript
// in renderer...
info.programs = programCache.programs;
```
? If we use a method on the cache, it may provoke the impression that the implementation can be changed, but it won't be the case unless we use `defineProperty` in the renderer...